Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -300,64 +315,139 @@ std::string LeptonRecoSFProducer::GetProducerId() const { return "LeptonRecoSFPr
 void LeptonRecoSFProducer::Init(ZJetSettings const& settings)
 {
     weightName = "RecoSFWeight";
-    m_sffile = settings.GetLeptonRecoSFRootfile();
-    if (settings.GetLeptonSFVariation() == true) {
-        LOG(INFO) << this->GetProducerId() << ": SF variation up and down one sigma";
-    }
-    if (settings.GetChannel() == "mm") {
-        histoname = settings.GetLeptonRecoSFHistogramName();
-    } else {
+    if (settings.GetChannel() != "mm") {
         LOG(ERROR) << "LeptonIsoSFProducer not implemented for this channel";
     }
 
-    LeptonSFProducer::Init(settings);
-
     m_year = settings.GetLeptonRecoSFYear();
-    LOG(INFO) << this->GetProducerId() << ": using year " << m_year << " for highPt SFs.";
+    LOG(INFO) << this->GetProducerId() << ": using year " << m_year << " for RECO SFs.";
 }
 
 std::tuple<float, float> LeptonRecoSFProducer::GetScaleFactorAndUnc(KLV const& lepton) const
 {
+    if (10.0 < lepton.p4.Pt() && 200 < lepton.p4.Pt()) {
+        return GetMediumPtEff(lepton);
+    }
     if ((lepton.p4.P() > 50.0 && std::abs(lepton.p4.Eta()) < 1.6) ||
         (lepton.p4.P() > 100.0 && std::abs(lepton.p4.Eta()) < 2.4)) {
         return GetHighPtEff(lepton);
     }
-    // else fall back to given histo at medium/low pt
-    return LeptonSFProducer::GetScaleFactorAndUnc(lepton);
+    // else return 1 +/- 0
+    return std::make_tuple(1.0, 0.0);
 }
 
-void LeptonRecoSFProducer::SetOverflowPtBin()
```

Comment:
Just for my understanding: So this method is not needed anymore, because it gets replaced by the medium scale factors? Couldn't we keep it, for the case that we need to extrapolate some scale factor at some point in the future again?

Reply:
I'll remove this method.

Reply:
Sorry, I removed it form the Derived LeptonRecoSF class, but it's still available in the base class, where the last pt bin is extended. So it's not gone.